pkg/controller/machineset/machineset_controller_test.go

```diff
@@ -17,196 +17,90 @@ limitations under the License.
-					t.Errorf("expected validation error did not occur")
-				}
-			},
-		},
-		{
-			name: "Create the MachineSet object and expect Reconcile to be called and the Machines to be created",
-			instance: &machinev1beta1.MachineSet{
-				ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "default"},
-				Spec: machinev1beta1.MachineSetSpec{
-					Replicas: &replicas,
-					Selector: metav1.LabelSelector{
-						MatchLabels: labels,
-					},
-					Template: machinev1beta1.MachineTemplateSpec{
-						ObjectMeta: metav1.ObjectMeta{
-							Labels: labels,
-						},
-					},
-				},
-			},
-			expectedRequest: reconcile.Request{NamespacedName: types.NamespacedName{Name: "foo", Namespace: "default"}},
-			// Verify machines are created and recreated after deletion
-			verifyFnc: func() {
-				machines := &machinev1beta1.MachineList{}
-
-				// TODO(joshuarubin) there seems to be a race here. If expectInt sleeps
-				// briefly, even 10ms, the number of replicas is 4 and not 2 as expected
-				expectInt(t, int(replicas), func(ctx context.Context) int {
-					if err := c.List(ctx, machines); err != nil {
-						return -1
-					}
-					return len(machines.Items)
-				})
-
-				// Delete a Machine and expect Reconcile to be called to replace it.
-				m := machines.Items[0]
-				if err := c.Delete(context.TODO(), &m); err != nil {
-					t.Errorf("error deleting machine: %v", err)
-				}
-				select {
-				case recv := <-requests:
-					if recv != expectedRequest {
-						t.Error("received request does not match expected request")
-					}
-				case <-time.After(timeout):
-					t.Error("timed out waiting for request")
-				}
+// var expectedRequest = reconcile.Request{NamespacedName: types.NamespacedName{Name: "foo", Namespace: "default"}}
 
-				// TODO (robertbailey): Figure out why the control loop isn't working as expected.
-				/*
-					g.Eventually(func() int {
-						if err := c.List(context.TODO(), machines); err != nil {
-							return -1
-						}
-						return len(machines.Items)
-					}, timeout).Should(gomega.BeEquivalentTo(replicas))
-				*/
-			},
-		},
-	}
-
-	for _, tc := range testCases {
-		t.Logf("Running %q testcase", tc.name)
-		func() {
-			if err := c.Create(context.TODO(), tc.instance); err != nil {
-				t.Errorf("error creating instance: %v", err)
-			}
+var _ = Describe("MachineSet Reconciler", func() {
+	namespace := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "ms-test"}}
 
-			defer func() {
-				err := c.Delete(context.TODO(), tc.instance)
-				if err != nil {
-					t.Logf("error deleting %v: %v", tc.instance, err)
-				}
+	BeforeEach(func() {
+		By("Creating the namespace")
+		Expect(k8sClient.Create(ctx, namespace)).To(Succeed())
+	})
 
-				select {
-				case recv := <-requests:
-					if recv != tc.expectedRequest {
-						t.Error("received request does not match expected request")
-					}
+	AfterEach(func() {
+		By("Deleting the namespace")
+		Expect(k8sClient.Delete(ctx, namespace)).To(Succeed())
+	})
```

Comment:
There's no GC running so you'll need to delete all Machines and MachineSets too, probably worth a helper function that does this (list all machines/machinesets, iterate through and delete). Doing this way means you don't have to clean up within your test IT's, just rely on the suite to clean everything

Reply:
👍